rest/utilities_testing.go

```diff
@@ -228,7 +228,24 @@ func (rt *RestTester) Bucket() base.Bucket {
 
 		rt.DatabaseConfig.SGReplicateEnabled = base.BoolPtr(rt.RestTesterConfig.sgReplicateEnabled)
 
-		_, err = rt.RestTesterServerContext.AddDatabaseFromConfig(*rt.DatabaseConfig)
+		if rt.useLeakyTBForDB {
+			// If using a leaky bucket, make sure it ignores attempted closures to avoid double closures panic
+			// due to closing both RestTesterServerContext and testBucket in rt.Close()
+			leakyBucket, isLeaky := base.AsLeakyBucket(testBucket)
+			if isLeaky {
+				leakyBucket.SetIgnoreClose(true)
+			} else {
+				panic("Using option useLeakyTBForDB but testBucket is not a leaky bucket")
```

Comment:
Some other areas of this function use `rt.tb.Fatalf` to abort the affected test, instead of failing the entire test suite - which seems preferable.